internal/provider/auth.go

```diff
@@ -270,11 +271,44 @@ func (l *AuthLoginCommon) init(d *schema.ResourceData) (string, map[string]inter
 	return path, params, nil
 }
 
-func (l *AuthLoginCommon) checkRequiredFields(d *schema.ResourceData, required ...string) error {
+type authDefault struct {
+	field string
+
+	// envVars will override defaultVal
+	envVars []string
```

Comment:
Might be a good idea to document how `envVars` are evaluated - are they ORd, order of precedence, etc.